#! /usr/bin/env python
# Task Long configuration

PACKAGE='floor_nav'
import roslib; roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator import *
from task_manager_lib.parameter_generator import *

gen = TaskParameterGenerator()
#       Name                    Type            Description
#       Default    Min   Max
gen.add("goal_x",        double_t, 0,    "X coordinate of destination",  0.)
gen.add("goal_y",        double_t, 0,    "Y coordinate of destination",  0.)
gen.add("goal_theta",   double_t,  0,    "Theta coordinate of destination", 0.)
gen.add("smart",         bool_t, 0,    "switch between the dumb and smart method", False)
gen.add("relative",     bool_t, 0,    "When true, the goal is relative to the current position",  False)
gen.add("k_v",           double_t, 0,    "Gain for velocity control",  5.0)
gen.add("k_alpha",       double_t, 0,    "Gain for angular control",  3.0)
gen.add("max_velocity",  double_t, 0,    "Max allowed velocity",  2.0)
gen.add("max_angular_velocity",  double_t, 0, "Max allowed angular velocity",  1.0)
gen.add("sigma2",       double_t,   0,  "coeff that influence the smoothness of path in smart mode", 0.5)
gen.add("dist_threshold",double_t, 0,    "Distance at which a the target is considered reached",  0.1)
gen.add("angular_threshold",double_t, 0,    "Angle in radian at which a the target is considered reached",  0.01)



exit(gen.generate(PACKAGE, "floor_nav", "TaskGotoPose"))

